sfxr.lua

```diff
@@ -850,28 +850,28 @@ end
 
 --- Synthesize the sound to a LÖVE SoundData instance.
 -- @tparam[opt=44100] SAMPLERATE rate the sampling rate
--- @tparam[opt=0] BITDEPTH depth the bit depth
+-- @tparam[opt=16] BITDEPTH depth the bit depth
 -- @tparam[opt] love.sound.SoundData sounddata a SoundData instance (will be
 -- created if not passed)
 -- @treturn love.sound.SoundData a SoundData instance
 -- @treturn int the number of written samples
 -- @raise "invalid sampling rate: x", "invalid bit depth: x"
 function sfxr.Sound:generateSoundData(rate, depth, sounddata)
     rate = rate or 44100
-    depth = depth or 0
+    depth = depth or 16
     assert(sfxr.SAMPLERATE[rate], "invalid sampling rate: " .. tostring(rate))
     assert(sfxr.BITDEPTH[depth] and depth, "invalid bit depth: " .. tostring(depth))
 
-    local tab, count = self:generateTable(rate, depth)
+    local tab, count = self:generateTable(rate)
```

Comment:
Why remove depth? Now it's unused, but the generateTable function still accepts it.

Reply:
I did this change a while ago, but if I remember correctly, `depth` parameter can be set to 0 (float), 8 (uint8) and 16 (uint16). LÖVE SoundData can only be created with floats (0 is the default depth).

You can revert this line and compare how demo sounds with and without this change.